### PR TITLE
GTW-9198: update IP address descriptions to accept ipv6 as well

### PR DESCRIPTION
**Pull request by katy-hughes-cko**
## Checklist

- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

File: abc_spec/components/schemas/Payments/Payment.yaml [hunk 1]
```diff
@@ -125,7 +125,7 @@ properties:
         allOf:
           - $ref: '#/components/schemas/PhoneNumber'
   payment_ip:
-    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 addresses.
+    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 & IPv6 addresses.
```
Comment (eoin-shaughnessy-cko) [inline abc_spec/components/schemas/Payments/Payment.yaml:128]:
```suggestion
    description: The IP address used to make the payment. Used by Checkout.com's risk engine to check the customer's IP address – only accepts IPv4 and IPv6 addresses.
```

File: abc_spec/components/schemas/Payments/PaymentRequest.yaml [hunk 2]
```diff
@@ -114,7 +114,7 @@ properties:
     maxLength: 255
     example: 'http://example.com/payments/fail'
   payment_ip:
-    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 addresses.
+    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 & IPv6 addresses.
```
Comment (eoin-shaughnessy-cko) [inline abc_spec/components/schemas/Payments/PaymentRequest.yaml:117]:
```suggestion
    description: The IP address used to make the payment. Used by Checkout.com's risk engine to check the customer's IP address – only accepts IPv4 and IPv6 addresses.
```

File: abc_spec/components/schemas/Payments/Payout.yaml [hunk 3]
```diff
@@ -123,7 +123,7 @@ properties:
     maxLength: 255
     example: 'http://example.com/payments/fail'
   payment_ip:
-    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 addresses.
+    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 & IPv6 addresses.
```
Comment (eoin-shaughnessy-cko) [inline abc_spec/components/schemas/Payments/Payout.yaml:126]:
```suggestion
    description: The IP address used to make the payment. Used by Checkout.com's risk engine to check the customer's IP address – only accepts IPv4 and IPv6 addresses.
```

File: nas_spec/changelog.md [hunk 4]
```diff
@@ -1,7 +1,8 @@
 # Changelog
 
 | Date       | Description of change                                                                                                   |
-|------------|-------------------------------------------------------------------------------------------------------------------------|
+| ---------- | ----------------------------------------------------------------------------------------------------------------------- |
+| 2023/04/03 | Updated payment_ip to support IPv6 addresses                                                                            |
```
Comment (eoin-shaughnessy-cko) [inline nas_spec/changelog.md:5]:
```suggestion
| 2023/04/03 | Updated `payment_ip` to support IPv6 addresses                                                                            |
```

File: nas_spec/components/schemas/HostedPayments/HostedPaymentsRequest.yaml [hunk 5]
```diff
@@ -28,7 +28,7 @@ properties:
     type: string
     format: ipv4
     maxLength: 45
-    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 addresses
+    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 & IPv6 addresses
```
Comment (eoin-shaughnessy-cko) [inline nas_spec/components/schemas/HostedPayments/HostedPaymentsRequest.yaml:31]:
```suggestion
    description: The IP address used to make the payment. Used by Checkout.com's risk engine to check the customer's IP address – only accepts IPv4 and IPv6 addresses.
```

**Review by jason-dantzer-cko — DISMISSED** [review]

File: nas_spec/components/schemas/PaymentLinks/PaymentLinksRequest.yaml [hunk 6]
```diff
@@ -26,7 +26,7 @@ properties:
     type: string
     format: ipv4
     maxLength: 45
-    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 addresses
+    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 & IPv6 addresses
```
Comment (eoin-shaughnessy-cko) [inline nas_spec/components/schemas/PaymentLinks/PaymentLinksRequest.yaml:29]:
```suggestion
    description: The IP address used to make the payment. Used by Checkout.com's risk engine to check the customer's IP address – only accepts IPv4 and IPv6 addresses.
```

File: nas_spec/components/schemas/Payments/Payment.yaml [hunk 7]
```diff
@@ -138,7 +138,7 @@ properties:
         allOf:
           - $ref: '#/components/schemas/PhoneNumber'
   payment_ip:
-    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 addresses.
+    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 & IPv6 addresses.
```
Comment (eoin-shaughnessy-cko) [inline nas_spec/components/schemas/Payments/Payment.yaml:141]:
```suggestion
    description: The IP address used to make the payment. Used by Checkout.com's risk engine to check the customer's IP address – only accepts IPv4 and IPv6 addresses.
```

File: nas_spec/components/schemas/Payments/PaymentDetails.yaml [hunk 8]
```diff
@@ -139,7 +139,7 @@ properties:
         allOf:
           - $ref: '#/components/schemas/PhoneNumber'
   payment_ip:
-    description:  The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 addresses.
+    description:  The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 & IPv6 addresses.
```
Comment (eoin-shaughnessy-cko) [inline nas_spec/components/schemas/Payments/PaymentDetails.yaml:142]:
```suggestion
    description: The IP address used to make the payment. Used by Checkout.com's risk engine to check the customer's IP address – only accepts IPv4 and IPv6 addresses.
```

File: nas_spec/components/schemas/Payments/PaymentRequest.yaml [hunk 9]
```diff
@@ -117,7 +117,7 @@ properties:
     maxLength: 1024
     example: 'http://example.com/payments/fail'
   payment_ip:
-    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 addresses.
+    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 & IPv6 addresses.
```
Comment (eoin-shaughnessy-cko) [inline nas_spec/components/schemas/Payments/PaymentRequest.yaml:120]:
```suggestion
    description: The IP address used to make the payment. Used by Checkout.com's risk engine to check the customer's IP address – only accepts IPv4 and IPv6 addresses.
```

File: abc_spec/components/schemas/PaymentLinks/PaymentLinksRequest.yaml [hunk 10]
```diff
@@ -26,7 +26,7 @@ properties:
     type: string
     format: ipv4
     maxLength: 45
-    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 addresses
+    description: The IP address used to make the payment. Used by our risk engine to check the customer's IP address – only accepts IPv4 & IPv6 addresses
```
[the remark this answers is not in the document]
Comment (katy-hughes-cko) [reply abc_spec/components/schemas/PaymentLinks/PaymentLinksRequest.yaml:29]: @eoin-shaughnessy-cko  thank you! What's the rule here, is it if there are multiple sentences then it should end with a full stop?

**Comment by castiel-ho-cko** [comment]
Hi @katy-hughes-cko, unfortunately **Hosted Payments Page** and **Payment Links** have not supported IPv6 yet, and will need additional changes to provide support. Suggest to revert the changes under `***_spec/components/schemas/HostedPayments/**` and `***_spec/components/schemas/PaymentLinks/**`

**Review by eoin-shaughnessy-cko — APPROVED** [review]

**Review by eoin-shaughnessy-cko — DISMISSED** [review]

**Review by jason-dantzer-cko — APPROVED** [review]

**Pull request merged**